vaadin-side-nav-flow-parent/vaadin-side-nav-testbench/src/main/java/com/vaadin/flow/component/sidenav/testbench/SideNavItemElement.java

```diff
@@ -53,17 +56,28 @@ public boolean isActive() {
         return hasAttribute("active");
     }
 
-    public String getLabel() {
-        final WebElement unnamedSlot = getWrappedElement().getShadowRoot()
-                .findElement(By.cssSelector("slot:not([name])"));
-        return (String) executeScript(
-                "return arguments[0].assignedNodes()[0].textContent;",
-                unnamedSlot);
+    @Override
+    public void click() {
```

Comment:
`navigate` would at least give you an idea what it does, with `click` you won't know if it triggers navigation or if it toggles the expanded state (or both?).

Maybe we should have `navigate` and `toggleExpanded`. The implementations could also be improved to check if the respective elements (anchor, toggle button) actually exist, and throw a meaningful exception otherwise. It looks like the methods for clicking the anchor / toggle button would currently just result in a script error.

Reply:
The thing is, `click` should do both. If an item is not navigable and has child items, a click should result in expanding the item. However, like I mentioned in the other thread, how the clicks are handled in the component needs to be debugged to be certain. It might be a bug. There is a test `clickNonNavigableParent_childNodesDisplayed` which checks this double behaviour. The reason I included both is to still have an API that covers the test cases, and also provide a direct way to handle navigation through the anchor.

Throwing a meaningful exception makes sense. There are enough lines that result in script errors already.